File: iphone-code/code/Stasher/STChildMissionCompleteCustomTableViewCell.m
//
//  STChildMissionCompleteCustomTableViewCell.m
//  Stasher
//
//  Created by bhushan on 27/11/14.
//  Copyright (c) 2014 OAB. All rights reserved.
//

#import "STChildMissionCompleteCustomTableViewCell.h"

@implementation STChildMissionCompleteCustomTableViewCell

- (void)awakeFromNib {
    // Initialization code
    
    [self.labelMissionInfo setTextColor:[UIColor stasherTextColor]];
    [self.labelMissionInfo setFont:[UIFont FontGothamRoundedBookWithSize:9.0f]];
    
    [self.labelMissionTitle setTextColor:[UIColor stasherTextColor]];
    [self.labelMissionTitle setFont:[UIFont FontGothamRoundedBoldWithSize:12.0f]];
    
    [self.buttonSendReward.titleLabel setFont:[UIFont FontGothamRoundedMediumWithSize:13.0f]];
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

- (IBAction)childMissionCompleteCellRequestRewardButtonPressed:(id)sender
{
    if ([self.delegate respondsToSelector:@selector(ChildMissionCompletedCellRequestRewardButtonPressedWithDict:)]) {
        [self.delegate ChildMissionCompletedCellRequestRewardButtonPressedWithDict:self.missionDict];
    }
}

@end

File: swift-code/Stasher/STChildMissionCompleteCustomTableViewCell.swift
//  Converted with Swiftify v1.0.6292 - https://objectivec2swift.com/
//
//  STChildMissionCompleteCustomTableViewCell.swift
//  Stasher
//
//  Created by bhushan on 27/11/14.
//  Copyright (c) 2014 OAB. All rights reserved.
//
import UIKit
protocol STChildMissionCompletedCellDelegate: NSObjectProtocol {
    func childMissionCompletedCellRequestRewardButtonPressed(withDict dict: [AnyHashable: Any])
}
class STChildMissionCompleteCustomTableViewCell: UITableViewCell {

    weak var delegate: STChildMissionCompletedCellDelegate?
    @IBOutlet var labelMissionInfo: UILabel!
    @IBOutlet var labelMissionTitle: UILabel!
    @IBOutlet var buttonReward: UIButton!
    @IBOutlet var buttonSendReward: UIButton!
    @IBOutlet var labelRewardAmount: UILabel!
    @IBOutlet var giftIconImgView: UIImageView!
    var missionDict = [AnyHashable: Any]()


    func awakeFromNib() {
        // Initialization code
        labelMissionInfo.textColor = UIColor.stasherText()
        labelMissionInfo.font = UIFont.fontGothamRoundedBook(withSize: 9.0)
        labelMissionTitle.textColor = UIColor.stasherText()
        labelMissionTitle.font = UIFont.fontGothamRoundedBold(withSize: 12.0)
        buttonSendReward.titleLabel?.font = UIFont.fontGothamRoundedMedium(withSize: 13.0)
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
        // Configure the view for the selected state
    }

    @IBAction func childMissionCompleteRequestRewardButtonPressed(_ sender: Any) {
        if delegate?.responds(to: #selector(self.ChildMissionCompletedCellRequestRewardButtonPressedWithDict)) {
            delegate?.childMissionCompletedCellRequestRewardButtonPressed(withDict: missionDict)
        }
    }
}